hudi-utilities/src/main/java/org/apache/hudi/utilities/HoodieClusteringJob.java

```diff
@@ -256,7 +279,16 @@ private int doScheduleAndCluster(JavaSparkContext jsc) throws Exception {
       LOG.info("The schedule instant time is " + instantTime.get());
       LOG.info("Step 2: Do cluster");
       Option<HoodieCommitMetadata> metadata = client.cluster(instantTime.get()).getCommitMetadata();
+      cleanAfterCluster(client);
       return UtilHelpers.handleErrors(metadata.get(), instantTime.get());
     }
   }
+
+  private void cleanAfterCluster(SparkRDDWriteClient client) {
+    client.waitForAsyncServiceCompletion();
+    if (client.getConfig().isAutoClean() && !client.getConfig().isAsyncClean()) {
```

Comment:
What kind of async table service do we want to wait for here?

Reply:
I think we need to trigge a sync clean if it is enabled.

Reply:
> What kind of async table service do we want to wait for here?

Asyn-cleanning and Async-archiving.

Reply:
> I think we need to trigge a sync clean if it is enabled.
IF isAsyncClean is enable, spark offline job will start an async-cleaning in prewrite like flink job. So if isAsyncClean is disable then add a synchronous cleanup

Reply:
> I think we need to trigge a sync clean if it is enabled.

IF isAsyncClean is enable, spark offline job will start an async-cleaning in prewrite like flink job. So if isAsyncClean is disable then add a synchronous cleanup

Reply:
We only need to add a sync cleaning of it is **enabled**, does that make sense to you?

Reply:
> We only need to add a sync cleaning of it is **enabled**, does that make sense to you?


I originally intended to maintain consistency with Flinkjob, retaining both cleaning way.  Adjusted to only keep synchronous cleaning mode